Drivers/CAN/can.c

```diff
@@ -0,0 +1,289 @@
+    if (CAN_Status!= HAL_OK)
+    {
+        return FAIL;
+    }
+    CAN_Status = HAL_CAN_Start(CAN3);
+    if (CAN_Status!= HAL_OK)
+    {
+        return FAIL;
+    }
+    
+    return SUCCESS;
+}
+
+CAN_t CAN_setup(uint32_t id, uint8_t len, uint8_t* data, uint8_t bus, CAN_HandleTypeDef* can_line)
+{
+    CAN_t CAN;
+    // Set up the message structure with ID and length of payload.
+    CAN.bus = bus;
+    uint8_t *data_temp;
+    for (int i = 0; i < 8; i++) 
+    {
+        if (i < len)
+        {
+            data_temp = const_cast<uint8_t*>(data + i);
+            CAN_message.data[i] = *data_temp;
+        }
+        else
+        {
+            CAN_message.data[i] = 0; // copies data to message, padding with 0s if length isn't 8
+        }
+    }
+    switch(bus)
+    {
+        case CAN_line_1:
+            CAN_message.CAN_handler = CAN1;
+            break;
+        case CAN_line_2:
+            CAN_message.CAN_handler = CAN2;
+            break;
+        case CAN_line_3:
+            CAN_message.CAN_handler = CAN3;
+            break;
+    }
+
+    CAN_TxHeaderTypeDef* tx_header = malloc(sizeof(CAN_TxHeaderTypeDef));
+    tx_header->StdId = id;
+    tx_header->ExtId = CAN_EXT_ID;
+    tx_header->IDE = CAN_ID_STD;
+    tx_header->RTR = CAN_RTR_data;
+    tx_header->DLC = len;
+    tx_header->TransmitGlobalTime = DISABLE;
+    CAN.CAN_tx_header = tx_header;
+
+    CAN_RxHeaderTypeDef* rx_header = malloc(sizeof(CAN_RxHeaderTypeDef));
+    rx_header->StdId = id;
+    rx_header->ExtId = CAN_EXT_ID;
+    rx_header->IDE = CAN_ID_STD;
+    rx_header->RTR = CAN_RTR_data;
+    rx_header->DLC = len;
+    rx_header->Timestamp = 0;
+    rx_header->FilterMatchIndex = 0;
+    CAN.CAN_rx_header = rx_header;
+    return CAN;
+}
+
+int sendMessageCAN1(uint32_t id, uint8_t len, const uint8_t *data)
+{
+    CAN_t message = CAN_setup(id, len, data, CAN_line_1, CAN1);
+    uint32_t free_spots = HAL_CAN_GetTxMailboxesFreeLevel(message.CAN_handler);
+    if(free_spots != 0)
+    {
+        HAL_StatusTypeDef bruh = HAL_CAN_AddTxMessage(message.CAN_handler, message.CAN_transmit_header, message.data, CAN_TX_MAILBOX0);
+        if(bruh == HAL_OK)
+        {
+            return SUCCESS;
+        }
+        else
+        {
+            return FAIL;
+        }
+    }
+    else
+    {
+        return FAIL;
+    }
+}
+
+int sendMessageCAN2(uint32_t id, uint8_t len, const uint8_t *data)
+{
+    CAN_t message = CAN_setup(id, len, data, CAN_line_2, CAN2);
+    uint32_t free_spots = HAL_CAN_GetTxMailboxesFreeLevel(message.CAN_handler);
+    if(free_spots != 0)
+    {
+        HAL_StatusTypeDef bruh = HAL_CAN_AddTxMessage(message.CAN_handler, message.CAN_transmit_header, message.data, CAN_TX_MAILBOX1);
+        if(bruh == HAL_OK)
+        {
+            return SUCCESS;
+        }
+        else
+        {
+            return FAIL;
+        }
+    }
+    else
+    {
+        return FAIL;
+    }
+}
+
+int sendMessageCAN3(uint32_t id, uint8_t len, const uint8_t *data)
```

Comment:
I still think we should try and avoid the ```sendCANMessage1()``` type functions and opt for passing in a CAN structure that contains the config and switch statement based on that.

Reply:
full heardily agree that we need to move away on all fronts from line specifc functions, shepherd has a ton of tech debt in that regard that i dislike